csm/core/services/audit_log.py

```diff
@@ -80,7 +82,7 @@ def _prepare_time_range(self, field, time_range: DateTimeRange):
         return db_conditions
 
     async def retrieve_by_range(self, component, limits,
-                       time_range: DateTimeRange):
+                       time_range: DateTimeRange, sort: Optional[str] = None):
```

Comment:
COPY & PASTE error : Signature of sort is not correct. You are passing an object SortBy and signature says string. 
Can you confirm whether this is the main reason why sorting is not working for StringType

Reply:
Ahh, my mistake, I completely missed the signature. I have changed it to SortBy and I confirmed StringType sorting is still not working. @pawankumarsrivastava